service/storage_proxy.cc

```diff
@@ -4071,6 +4071,12 @@ void storage_proxy::send_to_live_endpoints(storage_proxy::response_id_type respo
     auto& stats = handler_ptr->stats();
     auto& handler = *handler_ptr;
     auto& global_stats = handler._proxy->_global_stats;
+
+    if (handler.get_targets().size() == 0) {
+        remove_response_handler(response_id);
```

Comment:
What happens if hits are disabled? We remove the handler but I do not thin handler._ready will be ever resolved in this case.

Reply:
I took a glance at the code. Although it might happen that _ready promise is not explicitly set, the whole handler will be destroyed here so IIUC the write finish with a "broken promise" exception.

This reminds me of https://github.com/scylladb/scylladb/issues/12711. Even if hints are enabled (which is always the case for MV hints), if the target node is down long enough or runs out of space for hints then hints will not be written. Maybe this is a good explanation of the issue.

Reply:
yes, I tested this and saw the request complete after short time with "broken promise"
but I will address it to make it more explicit

Reply:
I changed it to check if CL is not achieved, then set _error
now when the handler is destroyed it sets _ready with the error